### ISSUE TITLE
Refactor "file_pack" resources using the datapackage interface?

### ISSUE THREAD

**Issue by manuelma** [Spine-project/Spine-Toolbox#1698]
I feel file_pack resources are the same concept as datapackage. The implementation of file_pack is a bit invasive in the sense we need to have a lot of if type == "file_pack" in different places. I wonder if it would be possible to refactor using a seamless method. I'm thinking we can at least move all those ifs to methods in ProjectItemResource that are called from the places where it's needed. In this way, it would be easier to see what's special about "file_pack".

- Show the list of files inside the pack in "Available resources" list views?
- Expand the pack into individual paths for passing command line arguments to the Tool?
- Something else?

**Comment by soininen** [comment]
I agree the file_pack stuff is a hack build on top of a system that was essentially designed for single files. It'd be great if we managed to improve it.
